src/libhirte/common/event-util.c

```diff
@@ -0,0 +1,131 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <systemd/sd-event.h>
+
+typedef uint64_t usec_t;
+
+static inline usec_t usec_add(usec_t a, usec_t b) {
+
+        if (a > UINT64_MAX - b) {
+                return UINT64_MAX;
+        }
+
+        return a + b;
+}
+
+static int event_reset_time(
+                sd_event *e,
+                sd_event_source **s,
+                clockid_t clock,
+                uint64_t usec,
+                uint64_t accuracy,
+                sd_event_time_handler_t callback,
+                void *userdata,
+                int64_t priority,
+                const char *description,
+                bool force_reset) {
+
+        bool created = false;
+        int enabled = 0;
+        int r = 0;
+        clockid_t c = 0;
+
+        assert(e);
+        assert(s);
+
+        if (*s) {
+                if (!force_reset) {
+                        r = sd_event_source_get_enabled(*s, &enabled);
+                        if (r < 0) {
+                                return r;
+                        }
+
+                        if (enabled != SD_EVENT_OFF) {
+                                return 0;
+                        }
+                }
+
+                r = sd_event_source_get_time_clock(*s, &c);
+                if (r < 0) {
+                        return r;
+                }
+
+                if (c != clock) {
+                        return r;
+                }
+
+                r = sd_event_source_set_time(*s, usec);
+                if (r < 0) {
+                        return r;
+                }
+
+                r = sd_event_source_set_time_accuracy(*s, accuracy);
+                if (r < 0) {
+                        return r;
+                }
+
+                (void) sd_event_source_set_userdata(*s, userdata);
+
+                r = sd_event_source_set_enabled(*s, SD_EVENT_ONESHOT);
+                if (r < 0) {
+                        return r;
+                }
+        } else {
+                r = sd_event_add_time(e, s, clock, usec, accuracy, callback, userdata);
+                if (r < 0) {
+                        return r;
+                }
+
+                created = true;
+        }
+
+        r = sd_event_source_set_priority(*s, priority);
+        if (r < 0) {
+                return r;
+        }
+
+        if (description) {
+                r = sd_event_source_set_description(*s, description);
+                if (r < 0) {
+                        return r;
+                }
+        }
+
+        return created;
```

Comment:
I'm not a fan of mixing return types. Furthermore, IIUC, you are counting on `true` to be a positive integer (which is implementation specific).
The current caller to this function does not care if the event was created or just rescheduled. So, first, think if you really need to return `created`. If not, just return 0. If yes, consider moving it to an OUT parameter (i.e. pointer to bool)

Reply:
I totally agree.  Fixed!